mesonbuild/msubprojects.py

```diff
@@ -520,8 +520,8 @@ def add_arguments(parser):
     p.set_defaults(subprojects_func=Runner.purge)
     p.set_defaults(post_func=Runner.post_purge)
 
-def run(options):
-    src_dir = os.path.relpath(os.path.realpath(options.sourcedir))
+def run(options: argparse.Namespace) -> int:
+    src_dir = os.path.relpath(os.path.realpath(T.cast(str, options.sourcedir)))
```

Comment:
There's a better option for this, you can write something like:
```python
if T.TYPE_CHECKING:
    from typing_extensions import Protocol

    class Arguments(Protocol):
         sourcedir: str
         types: str
         num_processes: bool

def run(options: 'Arguments):
    ...
```

Reply:
You may have to cast to the 'Arguments' somewhere, but it's still better in that once you do the single cast everything is correct.